src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,2490 @@
+        /// </summary>
+        [ThreadStatic]
+        private static bool t_cacheResourcesFromMultipleModuleVersions;
+
+        private static bool CacheResourcesFromMultipleModuleVersions
+        {
+            get
+            {
+                return t_cacheResourcesFromMultipleModuleVersions;
+            }
+
+            set
+            {
+                t_cacheResourcesFromMultipleModuleVersions = value;
+            }
+        }
+
+        /// <summary>
+        /// Flag shows if PS7 DSC APIs were used.
+        /// </summary>
+        [ThreadStatic]
+        public static bool NewApiIsUsed = false;
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        public static void Initialize()
+        {
+            Initialize(errors: null, modulePathList: null);
+        }
+
+        /// <summary>
+        /// Initialize the class cache with default classes that come with PSDesiredStateConfiguration module.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">List of module path from where DSC PS modules will be loaded.</param>
+        public static void Initialize(Collection<Exception> errors, List<string> modulePathList)
+        {
+            s_tracer.WriteLine("Initializing DSC class cache");
+
+            // Load the base schema files.
+            ClearCache();
+            var dscConfigurationDirectory = Environment.GetEnvironmentVariable("DSC_HOME");
+            if (string.IsNullOrEmpty(dscConfigurationDirectory))
+            {
+                var moduleInfos = ModuleCmdletBase.GetModuleIfAvailable(new Microsoft.PowerShell.Commands.ModuleSpecification()
+                    {
+                        Name = "PSDesiredStateConfiguration",
+
+                        // Version in the next line is actually MinimumVersion
+                        Version = new Version(3, 0, 0)
+                    });
+
+                if (moduleInfos.Count > 0)
+                { 
+                    // to be consistent with Import-Module behavior, we use the first occurrence that we find in PSModulePath
+                    var moduleDirectory = Path.GetDirectoryName(moduleInfos[0].Path);
+                    dscConfigurationDirectory = Path.Join(moduleDirectory, "Configuration");
+                }
+                else
+                {
+                    // when all else has failed use location of system-wide PS module directory (i.e. /usr/local/share/powershell/Modules) as backup
+                    dscConfigurationDirectory = Path.Join(ModuleIntrinsics.GetSharedModulePath(), "PSDesiredStateConfiguration", "Configuration");
+                }
+            }
+
+            if (!Directory.Exists(dscConfigurationDirectory))
```

Comment:
Any changing of the casing of the folder names will break Linux (and macOS if the File system is set to case sensitive)

Reply:
That is good to know this specifics of the API; thank you.
Though it feels like probability of this happening is rather low, especially considering that (apart from two substrings that are fixed/defined by the module) most of the path comes dynamically from PS APIs.